macros/light-sdk-macros/tests/hasher.rs

```diff
@@ -28,51 +29,392 @@ pub struct MyNestedNonHashableStruct {
+            };
+
+            let double = TruncatedStruct {
+                data: [2u8; 32], // All 2s
+            };
+
+            let mixed = TruncatedStruct {
+                data: {
+                    let mut data = [1u8; 32];
+                    data[0] = 2u8; // Change first byte
+                    data
+                },
+            };
+
+            let single_hash = single.hash::<Poseidon>().unwrap();
+
+            let double_hash = double.hash::<Poseidon>().unwrap();
+
+            let mixed_hash = mixed.hash::<Poseidon>().unwrap();
+
+            assert_ne!(single_hash, double_hash);
+            assert_ne!(single_hash, mixed_hash);
+            assert_ne!(double_hash, mixed_hash);
+        }
+
+        #[test]
+        fn test_truncation_longer_array() {
+            #[derive(LightHasher)]
+            struct LongTruncatedStruct {
+                #[truncate]
+                data: String,
+            }
+
+            let large_data = "a".repeat(64);
+            let truncated = LongTruncatedStruct {
+                data: large_data.clone(),
+            };
+
+            let mut modified_data = large_data.clone();
+            modified_data.push('b');
+            let truncated2 = LongTruncatedStruct {
+                data: modified_data,
+            };
+
+            let hash1 = truncated.hash::<Poseidon>().unwrap();
+            let hash2 = truncated2.hash::<Poseidon>().unwrap();
+
+            assert_ne!(hash1, hash2);
+        }
+
+        #[test]
+        fn test_multiple_truncates() {
+            #[derive(LightHasher)]
+            struct MultiTruncate {
+                #[truncate]
+                data1: String,
+                #[truncate]
+                data2: String,
+            }
+
+            let test_struct = MultiTruncate {
+                data1: "a".repeat(64),
+                data2: "b".repeat(64),
+            };
+
+            let hash1 = test_struct.hash::<Poseidon>().unwrap();
+
+            // Modifying data after truncation point shouldn't affect hash
+            let test_struct2 = MultiTruncate {
+                data1: "a".repeat(65),
+                data2: "b".repeat(65),
+            };
+
+            let hash2 = test_struct2.hash::<Poseidon>().unwrap();
+            assert_ne!(
+                hash1, hash2,
+                "Different data should produce different hashes"
+            );
+        }
+
+        #[test]
+        fn test_nested_with_truncate() {
+            #[derive(LightHasher)]
+            struct NestedTruncate {
+                #[nested]
+                inner: MyNestedStruct,
+                #[truncate]
+                data: String,
+            }
+
+            let nested = create_test_nested_struct();
+            let test_struct = NestedTruncate {
+                inner: nested,
+                data: "test".to_string(),
+            };
+
+            let hash = test_struct.hash::<Poseidon>().unwrap();
+            assert_eq!(hash.len(), 32);
```

Comment:
imo we should add a manual comparison as well to make sure that the combination of nested and truncated really works.